packages/core/src/internal/decorators/query-slot.ts

```diff
@@ -66,11 +72,19 @@ export function querySlot(selector: string, config?: QuerySlotConfig) {
  *
  * @ExportDecoratedItems
  */
-export function querySlotAll(selector: string) {
+export function querySlotAll(selector: string, config?: QuerySlotConfig) {
   return (protoOrDescriptor: {} | any, name?: PropertyKey): any => {
     const descriptor = {
       get(this: LitElement) {
-        return this.querySelectorAll(selector);
+        const ref = this.querySelectorAll(selector);
+
+        if (config?.assign) {
+          Array.from(ref)
+            .filter(i => !i.hasAttribute('slot'))
+            .forEach(i => i.setAttribute('slot', config.assign as string));
+        }
```

Comment:
Adds a `assign` option to query slot to easily assign a slotted element to a particular slot name